botocore/exceptions.py

```diff
@@ -342,17 +342,30 @@ class UnsupportedSignatureVersionError(BotoCoreError):
 class ClientError(Exception):
     MSG_TEMPLATE = (
         'An error occurred ({error_code}) when calling the {operation_name} '
-        'operation: {error_message}')
+        'operation{retry_info}: {error_message}')
 
     def __init__(self, error_response, operation_name):
+        retry_info = self._get_retry_info(error_response)
         msg = self.MSG_TEMPLATE.format(
             error_code=error_response['Error'].get('Code', 'Unknown'),
             error_message=error_response['Error'].get('Message', 'Unknown'),
-            operation_name=operation_name)
+            operation_name=operation_name,
+            retry_info=retry_info,
+        )
         super(ClientError, self).__init__(msg)
         self.response = error_response
         self.operation_name = operation_name
 
+    def _get_retry_info(self, response):
+        retry_info = ''
+        if 'ResponseMetadata' in response:
+            metadata = response['ResponseMetadata']
+            if metadata.get('MaxAttemptsReached', False):
+                if 'RetryAttempts' in metadata:
+                    retry_info = (' (max_retries=%s reached)' %
```

Comment:
This may be a little picky but from the context of the CLI ``max_retries`` seems pretty specific as if it is a parameter they can specify but it is not (it is not even something in botocore you can specify). I wonder if it makes sense to make this more like: ``' (reached max retry attempts: %s)'``?